articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,340 @@
+
+1. We initialization of the counter variable outside the syntax. 
+
+2. It begins with the condition check, if it's true, the code executes. 
+
+3. The value of the counter variable then increments or decrements, and the new counter variable goes through the condition check. 
+
+4. It will continue until the condition check is false.
+
+> NOTE: 
+> We must always have the increment or decrement part of the While Loop, failure to which will create a never-ending loop.
+
+***Example:***
+
+```Dart
+void main(){
+  int i = 0;
+   while(i < 3){
+      print("Hello");
+      i++;
+   }
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+```
+
+Our condition check begins with 0<3 which is true therefore the code executes and prints out the first hello. The value of i increases by one to 1 and the condition 1<3 is still true, so the code executes. It goes, the new variable becomes (2<3) and ends when the condition is now (3<3) which is false. Hence three hellos get printed. 
+
+#### Do-While Loop
+
+The do-While loop is an example of [Indefinite Loops](#indefinite-loops).
+
+**Syntax of Do-While Loop:**
+
+```Dart
+void main(){
+   // initialize counter variable
+
+   do {
+      //code;
+     
+      increment/decrement;
+   } while(condition);
+}
+```
+
+**How the Do-While Loop works:**
+
+**Code Execution - Increment - Condition Check**.
+
+1. Code executes then the counter variable initialized outside the syntax increments or decrements. 
+
+2. The condition check - if it's met, it begins from code execution. 
+
+3. The code execution will continue until the condition check turns false. 
+
+> Note that the code block in the Do-While loop executes at least once.
+
+***Example:***
+
+```Dart
+void main(){
+   int i = 0;
+   do {
+      print ("Hello");
+      i++;
+   } while (i<5);
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+
+The code first executes then the counter variable increments by one. It then checks the condition (1<5). It is true, so the loop will begin again. The code executes, then the counter variable increments by one. It then checks the condition (2<5). It meets the condition. It will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.
+
+### Other Key Concepts
+#### Break Keyword
+
+We use the `Break Keyword` to forcefully end a loop. It comes in handy when you want a partial output on a certain condtion and not to run all iterations.
+
+***Example:***
+
+```Dart
+void main(){
+ for (int i = 1; i<=10; i++){
+    print ("Hello");
+    if(i >= 6){
+       break;
+    }
+  }
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+This code will end the loop after printing five hellos because of the break statement.
+
+#### Continue Keyword
+
+We use the `Continue Keyword` to skip code or a loop iteration after the continuation and proceed with the rest of the iteration.
```

Comment:
`continue` must be lowercase.

Rewrite as: 

We use the 'continue` keyword ...